src/dune_engine/build_system.ml

```diff
@@ -1192,8 +1194,8 @@ end = struct
     let static_deps (type a) (t : a t) = Build.static_deps (build t)
 
     let evaluate_and_discover_dynamic_deps_unmemoized t =
-      let+ () = build_deps (static_deps t).rule_deps in
-      Build.exec (build t)
+      let* () = build_deps (static_deps t).rule_deps in
```

Comment:
```suggestion
      (* CR-someday amokhov: There is some code duplication here. The two lines below
         match the code that handles the `Build` case of `Build_exec.exec`. It would be
         nice to get rid of this little duplication. *)
      let* () = build_deps (static_deps t).rule_deps in
```

Reply:
Yes, sounds good, but let's call the function `build_deps_and_exec`, as otherwise it might seem that we're just computing them.